middleman-core/lib/middleman-core/util/data.rb

```diff
@@ -57,6 +57,7 @@ def parse(full_path, known_type=nil)
   # @return [Hash]
   Contract String, Pathname, Bool => Hash
   def parse_yaml(content, full_path)
+    content.sub!(/\\---/, '---')
```

Comment:
`parse_yaml` should only be called on valid YAML/JSON files (backslash makes it not valid YAML), so this shouldn't be here... are you calling `parse_yaml` directly from an extension or somewhere else?